queenbee/schema/workflow.py

```diff
@@ -0,0 +1,81 @@
+"""Queenbee workflow class.
+
+Workflow is a collection of operators and inter-related tasks that describes an end to
+end steps for the workflow.
+"""
+from queenbee.schema.qutil import BaseModel
+from pydantic import Schema, validator
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+import json
+import os
+from queenbee.schema.dag import DAG
+from queenbee.schema.arguments import Arguments
+from queenbee.schema.operator import Operator
+from queenbee.schema.function import Function
+
+
+class Workflow(BaseModel):
+    """A DAG Workflow."""
+    type: Enum('Workflow', {'type': 'Workflow'}) = 'Workflow'
+
+    name: str
+
+    id: str = str(uuid4())
+
+    inputs: Arguments
+
+    operators: List[Operator]
+
+    templates: List[Union[Function, DAG, 'Workflow']]
+
+    flow: DAG = Schema(
+        ...,
+        description='A list of steps for using tasks in a DAG workflow'
+    )
+
+    outputs: Arguments
+
+    # @validator('flow')
+    # def check_templates(cls, v, values):
+    #     """Check templates in flow to ensure they exist."""
+    #     template_names = set(t.name for t in values['templates'])
+    #     for task in v.tasks:
+    #         if task.template not in template_names:
+    #             raise ValueError('{} is not a valid template.'.format(task.template))
```

Comment:
I'd say it's generally bad practice to leave commented code in a commit. If you want to keep it here for future reference, I'd at least add a `TODO:` at the top of it.

Reply:
:police_car: :policeman: :police_car:

Reply:
Added a TODO and totally agreed!